File: exam-v2-frontend/src/api/score.ts
import http from './http'
import type { ApiResponse, PageResult } from './notice'

export interface ScoreRecord {
    scoreId: number
    examCode: number
    studentId: number
    subject: string
    ptScore: number | null
    etScore: number | null
    score: number | null
    answerDate: string
}

export function listScores(studentId: number, page: number, size: number) {
    return http.get<ApiResponse<PageResult<ScoreRecord>>>('/api/scores', {
        params: { studentId, page, size }
    })
}

export function getScoreById(id: number) {
    return http.get<ApiResponse<ScoreRecord>>(`/api/scores/${id}`)
}

File: exam-v2-backend/src/main/java/com/exam/entity/Score.java
package com.exam.entity;

/**
 * 成绩 — 对应 score 表
 */
public class Score {

    private Integer scoreId;
    private Integer examCode;
    private Integer studentId;
    private String subject;
    private Integer ptScore;
    private Integer etScore;
    private Integer score;
    private String answerDate;

    public Score() {
    }

    public Integer getScoreId() { return scoreId; }
    public void setScoreId(Integer scoreId) { this.scoreId = scoreId; }
    public Integer getExamCode() { return examCode; }
    public void setExamCode(Integer examCode) { this.examCode = examCode; }
    public Integer getStudentId() { return studentId; }
    public void setStudentId(Integer studentId) { this.studentId = studentId; }
    public String getSubject() { return subject; }
    public void setSubject(String subject) { this.subject = subject; }
    public Integer getPtScore() { return ptScore; }
    public void setPtScore(Integer ptScore) { this.ptScore = ptScore; }
    public Integer getEtScore() { return etScore; }
    public void setEtScore(Integer etScore) { this.etScore = etScore; }
    public Integer getScore() { return score; }
    public void setScore(Integer score) { this.score = score; }
    public String getAnswerDate() { return answerDate; }
    public void setAnswerDate(String answerDate) { this.answerDate = answerDate; }
}
